apps/website/src/composables/wallet.ts

```diff
@@ -1,47 +1,69 @@
 import { ref } from 'vue'
 import { ethers } from 'ethers'
 import { WalletProvider } from '@/interfaces/WalletProvider'
+import useIopay from '@/composables/iopay'
+
+const { getIoPayAccounts, getIoPayProvider, sendIoPayTransaction } = useIopay()
 
 const defaultProviders = {
   metamask: undefined,
   coinbase: undefined,
+  iopay: undefined,
 }
 
 export default function useWallet() {
   const ethereum: any = window.ethereum
   const availableProviders = ref<Record<string, WalletProvider>>(
     getAvailableProviders(ethereum)
   )
-  const selectedProvider = ref<WalletProvider>({})
+  const selectedProvider = ref<WalletProvider | string>({}) // Adding string type to support 'iopay' throws new ts errors in sendTransaction
```

Comment:
Make `selectedProvider` a ref<string>(''). 

"CoinbaseWallet", "MetaMask", "IoPay" can be the select strings you use (keep case consistent with the given provider map).

Reply:
Can you also update the `WalletProvider` interface to `EthersProvider`? You'll need to change the filename and interface export and import, but thinking it will clear things up as we add more.